packages/flutter_web_plugins/lib/src/plugin_event_channel.dart

```diff
@@ -0,0 +1,93 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+import 'plugin_registry.dart';
+
+/// A named channel for sending events to the framework-side using streams.
+///
+/// This is the platform-side equivalent of [EventChannel]. Whereas
+/// [EventChannel] receives a stream of events from platform plugins, this
+/// channel sends a stream of events to the handler listening on the
+/// framework-side.
+class PluginEventChannel<T> {
+  /// Creates a new plugin event channel.
+  const PluginEventChannel(
+    this.name, [
+    this.codec = const StandardMethodCodec(),
+    BinaryMessenger binaryMessenger,
+  ])  : assert(name != null),
+        assert(codec != null),
+        _binaryMessenger = binaryMessenger;
+
+  /// The logical channel on which communication happens, not null.
```

Comment:
I would move the phrase not-null to a new line, since it is incidental.

```
/// The logical channel on which communication happens.
///
/// This value will (must?) never be null.
```

Reply:
Done.